### PR TITLE
MOBT-494: Cube title setting in weather symbol code

### PR DESCRIPTION
**Pull request by bayliffe**
Adds the option to manually set the title attribute for weather symbols output. This enables the attribute to be set as desired, including in situations where the input cubes do not share a common title, resulting in no title being set otherwise.

This change is needed for the site-specific phase change work. In the new chains that produce weather symbols from spot forecasts, the rain / sleet / snow inputs have been spot-extracted prior to realization collapse and cycle blending. The weighted blending plugin inherits the PostProcessingPlugin class, meaning all titles are prefixed "Post processed". As a result:
- usual spot extracted outputs used as inputs to the wx have the title "MOGREPS-G Spot Values"
- the phase diagnostics have used as inputs to the wx have the title "Post-Processed MOGREPS-G Spot Values"
- the generate_mandatory_attributes code returns a default unset title attribute as the title is no longer common between all inputs.

This change will be accompanied by an app config change in the suite to manually set the weather symbols output title for the spot-forecast generated weather symbols.

Acceptance test data: https://github.com/metoppv/improver_test_data/pull/15

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by bayliffe** [comment]
@MoseleyS KGO title updated to be metadata compliant.

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks @bayliffe 👍 

I've just added a minor query related to the checksum.

File: improver_tests/acceptance/SHA256SUMS [hunk 1]
```diff
@@ -845,6 +845,7 @@ ea67ae7a7f5363ae3692b29c93fb9989449d96e49dc613a1297d98ddb12c578a  ./wxcode-modal
 803b2a645f8a33db4601a45dc16325f8a90f13fdea430b98ea94f032ee032180  ./wxcode/bad_wx_decision_tree.json
 c2fdb1144abdf6188bb2d9de0104c6dfbcb3a8ba07cbe4e7ac2b9c3782f55484  ./wxcode/basic/kgo.nc
 9c543a6e2068ad51c6664903bf06b5ecbf018ad914f972762ef0577469fec1b3  ./wxcode/basic/kgo_no_lightning.nc
+a51691b4b7b6f8661804554eecf47440ec8353c80d9a779d06ea4b0effbd5b42  ./wxcode/basic/kgo_titled.nc
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver_tests/acceptance/SHA256SUMS]: Yeah, no idea what happened there. Checksums recreated and pushed.

**Review by gavinevans — APPROVED** [review]

**Review by brhooper — APPROVED** [review]
This all looks fine, tests pass.

**Pull request merged**